src/basic/list.h

```diff
@@ -137,32 +137,32 @@
         (!(item)->name##_prev && !(item)->name##_next)
 
 #define LIST_FOREACH(name,i,head)                                       \
-        for ((i) = (head); (i); (i) = (i)->name##_next)
+        for (typeof(head) i = (head); i; i = i->name##_next)
```

Comment:
Can we use the `UNIQ_T` stuff here?

Reply:
No, the iterator must be usable in the loop. Hence, the provided name must be used.

Reply:
I see - but then nesting won't work?

Reply:
Why? There is no variable implicitly declared. We can safely use the macro arbitrary level of nesting:
```
LIST_FOREACH(name1, i, head1)
    LIST_FOREACH(name2, j, head2)
        ...;
```

Reply:
ah yes, I misread it, sorry